en/lessons/libraries/stream-data.md

```diff
@@ -0,0 +1,397 @@
+---
+version: 1.0.0
+title: StreamData
+---
+
+An example-based unit testing library like [ExUnit](https://hexdocs.pm/ex_unit/ExUnit.html) is a wonderful tool to help you verify that your code works the way you think it does. However, example-based unit tests have some drawbacks:
+
+* It can be easy to miss edge cases, since you're only testing a few inputs.
+* You can write these tests without thinking through your requirements thoroughly.
+* These tests can be very verbose when you use several examples for one function.
+
+In this lesson we're going to explore how [StreamData](https://github.com/whatyouhide/stream_data) can help overcome some of these drawbacks.
+
+
+{% include toc.html %}
+
+
+## What is StreamData?
+
+[StreamData](https://github.com/whatyouhide/stream_data) is library that performs stateless property-based testing.
+
+The StreamData library will run each test 100 times with random data. When a test fails, StreamData will try to shrink the input to the smallest value that causes the test failure. This can be helpful when you have to debug your code! If a 50-element list causes your function to break, and only one of the list elements is problematic, StreamData can help you identify the offending element.
+
+This testing library has two main modules. [`StreamData`](https://hexdocs.pm/stream_data/StreamData.html) generates streams of random data. [`ExUnitProperties`](https://hexdocs.pm/stream_data/ExUnitProperties.htm) lets you run tests against your functions, using the generated data as your input.
+
+You might be asking how you can say anything meaningful about a function if you don't know what your exact inputs are. Read on!
+
+
+## Using StreamData
```

Comment:
Can we add a bit on "getting started" that shoes you have to include the library in your mix file's dependencies?